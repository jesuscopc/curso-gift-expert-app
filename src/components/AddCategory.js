import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorie }) => {

  const [inputValue, setInputValue] = useState('');
  
  const handleChangeInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputValue.trim().length > 2) {
      setCategorie( [inputValue] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <h1>{ inputValue }</h1>
      <input
        type="text"
        value={ inputValue }
        onChange={ handleChangeInputValue }
      />
    </form>
  )
};

AddCategory.propTypes = {
  setCategorie: PropTypes.func.isRequired,
};