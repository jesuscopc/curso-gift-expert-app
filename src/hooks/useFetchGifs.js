import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGif";

export const useFetchGifs = ( categorie ) => {

  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {// Lo usamos para que no llamemos codigo innecesario cuanso se haga un re-render de todo el componente
    getGifs( categorie ).then(imgs => {
      setState({
          data: imgs,
          loading: false
        });
      });
    }, [ categorie ]); //Es un array de dependencias que dispararan la funcion dentro del useEffect
  
    return state;
 
}