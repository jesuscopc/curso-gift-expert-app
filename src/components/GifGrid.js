import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categorie }) => {

  const { data: imgs, loading } = useFetchGifs( categorie );

  return (
    <>
      <h3 className="card animate__animated animate__fadeIn">{ categorie }</h3>
      { loading && <p className="card animate__animated animate__flash">loading...</p> }
      <div className="card-grid">
          {imgs.map( img =>
            (
              <GifGridItem
                key={ img.id }
                { ...img } 
              />
            )
          )}
      </div>
    </>
  )
}
