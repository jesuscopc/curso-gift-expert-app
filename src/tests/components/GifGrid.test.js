import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Sui test GifGrid Component', () => {

  const categorie= 'Death Note';
  
  test('should be render correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid categorie={categorie} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render elements mocked', () => {
    const gifs = [{
      id: 'ABC',
      url: 'http://localhost/ciualquier/cosa.jpg',
      title: 'Cualquier cosa'
    }, {
      id: 'ABC',
      url: 'http://localhost/ciualquier/cosa.jpg',
      title: 'Cualquier cosa'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid categorie={categorie} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });



})
