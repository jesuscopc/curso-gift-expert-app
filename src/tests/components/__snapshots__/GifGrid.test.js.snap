// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui test GifGrid Component should be render correctly 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    Death Note
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Sui test GifGrid Component should be render elements mocked 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    Death Note
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/ciualquier/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/ciualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
